### ISSUE TITLE
setAngle() does not update bbox

### ISSUE THREAD

**Issue by slickgotchi** [Prozi/detect-collisions#61]
Hi,

I was using a long box and rotating it as a collider. 

Unlike setPosition(), when using setAngle() the bounding box does not appear to get updated for a collider and thus can get missed when using checkAll() and checkOne(). For my app I have to call system.insert() manually whenever I rotate my box collider. 

May need to create a setAngle() over-ride in the Polygon class.

Cheers!

**Comment by Prozi** [comment]
@slickgotchi thank you for opening an issue!

this is actually **intended behaviour** 🥲 

consider this case, you need to set angle, set position, and set rotation

if you would recreate bbox that would be 3x the effort each frame

that's why the `setPosition` function is unique in this way that is also updates the bbox by reinserting to system.

you should first set angle, set rotation, and then set position

also **if you don't need to set position just use `polygon.updateBody()` and not `system.insert(polygon)`**

**Comment by Prozi** [comment]
@slickgotchi I have rethought the issue and the architecture of manipulation functions and I think you're right

all of them should behave simiarly

so I have fixed this issue since v8.1.0 and you can read more about it here

https://github.com/Prozi/detect-collisions/releases/tag/v8.1.0

**Comment by Prozi** [comment]
TL;DR:

all functions don't update the bbox now, just mark body as dirty, and in next check* function it will update

**Comment by Prozi** [comment]
@slickgotchi ajajaj that v8.1.0 move was kind of a mistake

please use `body.updateBody()` on body you moved or

`system.update()` once per frame

to update bodies

also please use v9.0.0 as there is a bug in v8.1.0 more info here

https://github.com/Prozi/detect-collisions/releases/tag/v9.0.0

**Comment by Prozi** [comment]
I will still work out a compromise not to do this breaking change after all, gimme an hour top

**Comment by Prozi** [comment]
@slickgotchi

actually https://github.com/Prozi/detect-collisions/releases/tag/v9.1.0 this has what you requested

now all manipulation functions if not specified the parameter update is true so they update bbox not only setAngle but also setOffset and setScale now work like setPosition did before your issue.

that took me some reiterations didn't it, hope you enjoy

best regards

**Comment by slickgotchi** [comment]
Hey mate, sorry didn't mean to set you off down the rabbit hole but thanks so much for incorporating this!

Really enjoying this library and its speed/simplicity.

**Comment by Prozi** [comment]
no problem glad I could help
